cmd/syncthing/gui.go

```diff
@@ -189,16 +189,18 @@ func (s *apiService) getListener(guiCfg config.GUIConfiguration) (net.Listener,
 		},
 	}
 
-	rawListener, err := net.Listen("tcp", guiCfg.Address())
-	if err != nil {
-		return nil, err
-	}
-
-	listener := &tlsutil.DowngradingListener{
-		Listener:  rawListener,
-		TLSConfig: tlsCfg,
+	listeners := []net.Listener{}
+	for _, addr := range guiCfg.Addresses() {
+		rawListener, err := net.Listen("tcp", addr)
+		if err != nil {
+			return nil, err
+		}
+		listeners = append(listeners, &tlsutil.DowngradingListener{
+			Listener:  rawListener,
+			TLSConfig: tlsCfg,
```

Comment:
Feature request to specify a separate TLS certificate per listener incoming in 3, 2, 1...

Reply:
I see your point. Do you expect this feature as part of this pull request?

Reply:
Not necessarily, just throwing it out there in case you are willing to add that.

Reply:
I will investigate that point.

Reply:
One of the things I thought while glancing at the config diff was that the addition of `RawAddresses []string` might instead have been a change to have a slice of GUI configurations instead of just a single one. I'm not sure if this makes it more or less difficult.